application/crud.py

```diff
@@ -0,0 +1,93 @@
+from typing import List, Optional
+
+from sqlalchemy.orm import Session
+from fastapi import HTTPException, status
+
+from application import schemas
+from database import models
+
+
+def get_all_cities(db_session: Session) -> List[models.DBCity]:
+    cities = db_session.query(models.DBCity).all()
+    return cities
+
+
+def get_city_by_name(name: str, db_session: Session) -> schemas.CityInfo | None:
```

Comment:
The return type annotation `schemas.CityInfo | None` is not compatible with Python versions below 3.10. If you are using a version below 3.10, consider using `Optional[schemas.CityInfo]` instead.